src/services/jsDoc.ts

```diff
@@ -142,33 +153,35 @@ namespace ts.JsDoc {
     }
 
     function getCommentDisplayParts(tag: JSDocTag, checker?: TypeChecker): SymbolDisplayPart[] | undefined {
-        const { comment } = tag;
-        switch (tag.kind) {
+        const { comment, kind } = tag;
+        switch (kind) {
             case SyntaxKind.JSDocImplementsTag:
-                return withNode((tag as JSDocImplementsTag).class);
+                return withNode((tag as JSDocImplementsTag).class, syntaxKindDisplayPartFunctionDict[kind]);
             case SyntaxKind.JSDocAugmentsTag:
-                return withNode((tag as JSDocAugmentsTag).class);
+                return withNode((tag as JSDocAugmentsTag).class, syntaxKindDisplayPartFunctionDict[kind]);
             case SyntaxKind.JSDocTemplateTag:
-                return addComment((tag as JSDocTemplateTag).typeParameters.map(tp => tp.getText()).join(", "));
+                return addComment((tag as JSDocTemplateTag).typeParameters.map(tp => tp.getText()).join(", "), syntaxKindDisplayPartFunctionDict[kind]);
             case SyntaxKind.JSDocTypeTag:
-                return withNode((tag as JSDocTypeTag).typeExpression);
+                return withNode((tag as JSDocTypeTag).typeExpression, syntaxKindDisplayPartFunctionDict[kind]);
             case SyntaxKind.JSDocTypedefTag:
             case SyntaxKind.JSDocCallbackTag:
             case SyntaxKind.JSDocPropertyTag:
             case SyntaxKind.JSDocParameterTag:
             case SyntaxKind.JSDocSeeTag:
-                const { name } = tag as JSDocTypedefTag | JSDocPropertyTag | JSDocParameterTag | JSDocSeeTag;
-                return name ? withNode(name) : comment === undefined ? undefined : getDisplayPartsFromComment(comment, checker);
+                const { name } = tag as JSDocTypedefTag | JSDocCallbackTag | JSDocPropertyTag | JSDocParameterTag | JSDocSeeTag;
+                return name ? withNode(name, syntaxKindDisplayPartFunctionDict[kind]) : comment === undefined ? undefined : getDisplayPartsFromComment(comment, checker);
             default:
                 return comment === undefined ? undefined : getDisplayPartsFromComment(comment, checker);
         }
 
-        function withNode(node: Node) {
-            return addComment(node.getText());
+        function withNode(node: Node, constructDisplayPartFunc?: (text: string) => SymbolDisplayPart) {

```

Comment:
Might as well use closure to access the displayPart function for the tagName.

```suggestion
        function withNode(node: Node) {
```